CAN.cpp

```diff
@@ -0,0 +1,147 @@
+#include "CAN.h"
+#include "CONFIG.h"
+
+//——————————————————————————————————————————————————————————————————————————————
+//  ACAN2515 Demo in loopback mode, for the Raspberry Pi Pico
+//  Thanks to Duncan Greenwood for providing this sample sketch
+//——————————————————————————————————————————————————————————————————————————————
+
+#ifndef ARDUINO_ARCH_RP2040
+#error "Select a Raspberry Pi Pico board"
+#endif
+
+//——————————————————————————————————————————————————————————————————————————————
+
+#include <ACAN2515.h>
+
+//——————————————————————————————————————————————————————————————————————————————
+// The Pico has two SPI peripherals, SPI and SPI1. Either (or both) can be used.
+// The are no default pin assignments so they must be set explicitly.
+// Testing was done with Earle Philhower's arduino-pico core:
+// https://github.com/earlephilhower/arduino-pico
+//——————————————————————————————————————————————————————————————————————————————
+
+// This code is just the ACAN2515 LoopBackDemoRaspberryPiPico example code
+
+// Using SPI
+static const byte MCP2515_SCK  = 18; // SCK input of MCP2515
+static const byte MCP2515_MOSI = 19; // SDI input of MCP2515
+static const byte MCP2515_MISO = 16; // SDO output of MCP2517
+
+static const byte MCP2515_CS  = 17;  // CS input of MCP2515 (adapt to your design)
+static const byte MCP2515_INT = 15;  // INT output of MCP2515 (adapt to your design)
+
+//——————————————————————————————————————————————————————————————————————————————
+//  MCP2515 Driver object
+//——————————————————————————————————————————————————————————————————————————————
+
+ACAN2515 can (MCP2515_CS, SPI, MCP2515_INT) ;
+
+//——————————————————————————————————————————————————————————————————————————————
+//  MCP2515 Quartz: adapt to your design
+//——————————————————————————————————————————————————————————————————————————————
+
+static const uint32_t QUARTZ_FREQUENCY = 16UL * 1000UL * 1000UL; // 16 MHz
+
+//——————————————————————————————————————————————————————————————————————————————
+//   SETUP
+//——————————————————————————————————————————————————————————————————————————————
+
+void CAN_set(){
+  //--- Switch on builtin led
+  pinMode (LED_BUILTIN, OUTPUT);
+  digitalWrite (LED_BUILTIN, HIGH);
+  
+  //--- Start serial
+  Serial.begin (115200);
+  
+  //--- Wait for serial (blink led at 10 Hz during waiting)
+  while (!Serial) {
+    delay (50);
+    digitalWrite (LED_BUILTIN, !digitalRead (LED_BUILTIN)) ;
+  }
+  
+  //--- There are no default SPI pins so they must be explicitly assigned
+  SPI.setSCK(MCP2515_SCK);
+  SPI.setTX(MCP2515_MOSI);
+  SPI.setRX(MCP2515_MISO);
+  SPI.setCS(MCP2515_CS);
+  
+  //--- Begin SPI
+  SPI.begin ();

```

Comment:
We should consider moving SPI.setXXX and SPI.begin to icu.ino (ICU_Arduino.ino) in the setup() function. It will help us visibly see/debug our initializations.